### PR TITLE
Implementation of `multi_image_photometry()` code in photometry.py

### PR DESCRIPTION
**Pull request by JuanCab**
So this is the next big update, I wrote a function `multi_image_photometry()` that is essentially a wrapper of `single_image_photometry()`.  This will replace `photometry_on_directory()`. 

Some changes of note since the last PR:

- Cleaned up `single_image_photometry()` to 
	- output `None, None` if it fails but it's OK to proceed to the next time.  Exceptions are now only used when we really need to stop execution.
	- Fixed logic to find x/y positions from the WCS before attempting to filter out of bounds sources.
	- Check for the presence of require headers in CCDData before attempting aperture photometry.
	- Catch all the repeated warnings thrown by the fitter in `compute_fwhm()` and just return a summary of the number of failed fits to the screen.
- Fixed `compute_fwhm()` (really the 2D Gaussian fitter it calls) to handle NaN, which allows us to use NaN for high pixels. (Fixing issue #142)
- I added a test suite that creates multiple CCDData objects with shifted positions to test the new function.

When run on a file in a directory, the output looks something like:
```
multi_image_photometry: Processing image TIC_467615239.01-S001-R001-C001-ip.fit
  Calling single_image_photometry ...
    > 12 of 926 sources within 2 aperture radii of nearest neighbor ... removed them.
    > 50 sources too close to image edge ... removed them.
    > 864 of 926 original sources to have photometry done.
    > Computing clipped sky stats ... DONE.
    > Fitting FWHM of all sources (may take a few minutes) ... fitting failed on 119 of 864 sources  ... DONE.
    > Bad FWHM values (<1 pixel) for 244 sources.
    > Aperture net counts negative for 1 sources.
    > 245 sources with either bad FWHM fit or bad aperture net counts had aperture_net_cnts set to NaN.
    > Calculating noise for all sources ... DONE.
  Done with single_image_photometry for TIC_467615239.01-S001-R001-C001-ip.fit
```

I am hoping this is much cleaner and easier to follow without all the warning messages.

### CODE REVIEW

**Comment by mwcraig** [comment]
I think using the `ignore_cleanup_errors=False` option to `TemporaryDirectory` will stop the windows failure.

**Comment by JuanCab** [comment]
Just submitted a commit to add the ` ignore_cleanup_errors=False` to `TemporaryDirectory`.  It seems to have done the trick.

File: stellarphot/photometry/photometry.py [hunk 1]
```diff
@@ -154,6 +158,11 @@ def single_image_photometry(ccd_image, sourcelist, camera, observatory_location,
     the `use_coordinates` parameter should be set to "sky".
     """
 
+    def msgNexit(msg):
```
Comment (mwcraig) [inline stellarphot/photometry/photometry.py:161]: I'm not 100% convinced that this function is need but it should definitely be outside the `single_image_photometry` body and be underscore-named e.g. `msg_and_exit`

**Review by mwcraig — CHANGES_REQUESTED** [review]
There are a couple of minor changes inline.

More broadly, I've opened issues for:

- Figuring out how to better handle messages. We need some way to opt out of (or opt into) t verbose messaging. We might want to implement it using python logging. #148 
- Simplifying the function signatures. #149
- Seeing if we can simplify the test image classes a bit. Not a real high priority.

File: CHANGES.rst [hunk 2]
```diff
@@ -1,14 +1,16 @@
-1.3.10 (unreleased)
+1.4.0 (unreleased)
```
[the remark this answers is not in the document]
Comment (JuanCab) [reply CHANGES.rst:1]: I was gonna do that but didn't want to be presumptuous... how Minnesotan of me.

File: CHANGES.rst [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 ------------------
 
 New Features
 ^^^^^^^^^^^^
-+ Creation of new data classes for handling aperture, photometry, and catalog data in a more consistent way by enforcing validation and certain column names. [#125]
++ Development of new data classes for handling source list, photometry, and catalog data which include data format validation. [#125]
++ Aperture photometry streamlined into single_image_photometry() and multi_image_photometry() functions that use the new data classes. [#141]
```
[the remark this answers is not in the document]
Comment (JuanCab) [reply CHANGES.rst:7]: Done.

File: CHANGES.rst [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
++ multi_image_photometry() now is a wrapper for single_image_photometry() instead of a completely separate function.
 
 Other Changes and Additions
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 + Major reorganizaiton of code including moving functions to new modules. [#130, #133]
-
++ Now requires python 3.10 or later.
```
[the remark this answers is not in the document]
Comment (JuanCab) [reply CHANGES.rst:13]: Done.

File: stellarphot/photometry/photometry.py [hunk 5]
```diff
@@ -312,25 +346,12 @@ def single_image_photometry(ccd_image, sourcelist, camera, observatory_location,
...
-
-    try:
-        photom['date-obs'] = Column(data=Time([ccd_image.header['DATE-OBS']]
-                                              * len(photom),
-                                    format='isot', scale='utc'), name='date-obs')
-    except KeyError:
-        raise ValueError("'DATE-OBS' not found in CCD image header!")
+    # Set various columns based on CCDData headers (which we
+    # checked for earlier)
+    photom['exposure'] = [exposure] * len(photom) * u.second
```
[the remark this answers is not in the document]
Comment (JuanCab) [reply stellarphot/photometry/photometry.py:351]: I don’t believe FITS has a standard way of encoding units, so reliably extracting the units of the exposure time would be tricky.  I do *assume* it is in seconds.  None of the astropy documentation I just searched through shows examples of extracting units from FITS headers.   I have now noted that exposure must be in seconds in the documentation.

File: stellarphot/photometry/tests/test_photometry.py [hunk 6]
```diff
@@ -333,4 +337,201 @@ def test_aperture_photometry_with_outlier_rejection(reject):
...
+def list_of_fakes(num_files):
+    # Generate fake CCDData objects for use in photometry_on_directory tests
+    fake_images = [FakeCCDImage()]
+
+    # Create additional images, each in a different position.
+    for i in range(num_files-1):
+        angle = 2*np.pi/(num_files-1) * i
+        rad = 50
+        dx, dy = rad*np.cos(angle), rad*np.sin(angle)
+        fake_images.append( shift_FakeCCDImage(fake_images[0], dx, dy) )
```
[the remark this answers is not in the document]
Comment (JuanCab) [reply stellarphot/photometry/tests/test_photometry.py:352]: Fixed.

File: stellarphot/photometry/photometry.py (hunk 1 above)
Comment (JuanCab) [reply stellarphot/photometry/photometry.py:161]: I removed the inner function, I suppose it is a matter of style.

File: CHANGES.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (JuanCab) [reply CHANGES.rst:13]: Fixed.

File: stellarphot/photometry/photometry.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (JuanCab) [reply stellarphot/photometry/photometry.py:351]: After text discussion, ignoring this issue for now.

**Pull request merged**